doc/examples/affine_registration_masks.py

```diff
@@ -0,0 +1,363 @@
+"""
+==============================
+Affine Registration with Masks
+==============================
+
+This example explains how to compute a transformation to register two 3D
+volumes by maximization of their Mutual Information [Mattes03]_. The
+optimization strategy is similar to that implemented in ANTS [Avants11]_.
+
+We will use masks to define which pixels are used in the Mutual Information.
+Masking can also be done for registration of 2D images rather than 3D volumes.
+
+Masking for registration is much more necessary for cardiac MRI than for brain
+MRI, where it is usually used to specify a region of interest on a 2D static
+image, e.g. the left ventricle in a short axis slice. The prioritizes
+registering the region of interest over other structures that move with respect
+to the heart.
+
+"""
+
+from os.path import join as pjoin
+import numpy as np
+import matplotlib.pyplot as plt
+from dipy.viz import regtools
+from dipy.data import fetch_stanford_hardi
+from dipy.data.fetcher import fetch_syn_data
+from dipy.io.image import load_nifti
+from dipy.align.imaffine import (transform_centers_of_mass,
+                                 AffineMap,
+                                 MutualInformationMetric,
+                                 AffineRegistration)
+from dipy.align.transforms import (TranslationTransform3D,
+                                   RigidTransform3D,
+                                   AffineTransform3D)
```

Comment:
pep8: `AffineTransform3D` not used

Reply:
I've removed it now, and also some other imports that were not needed